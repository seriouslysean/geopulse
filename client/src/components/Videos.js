import React from "react";
import Loading from "../components/Loading";
import VideosVideo from "./VideosVideo";

class Videos extends React.Component {
  renderContent = () => {
    return (
      <div className="content">
        <ul className="videos__list">
          {this.props.videos.collection.map(video => {
            return <VideosVideo key={video.id.videoId} video={video} />;
          })}
        </ul>
      </div>
    );
  };
  render() {
    return (
      <section className="section section--spacing section--bg-earth-darker videos">
        <div className="content-container content-container--grid-spacing">
          <div className="videos__content">
            <h1 className="section__title">Videos</h1>

            {this.props.ready || this.props.videos ? this.renderContent() : <Loading />}

            <a rel="external" href="https://www.youtube.com/" className="section__attribution" target="_blank">
              Powered by YouTube
            </a>
          </div>
        </div>
      </section>
    );
  }
}

export default Videos;
