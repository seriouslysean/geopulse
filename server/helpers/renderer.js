import React from 'react';
import serialize from 'serialize-javascript';
import { renderToString } from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { renderRoutes } from 'react-router-config';
import { Helmet } from 'react-helmet';
import AppRoutes from '../../client/src/routes/AppRoutes';

export default (req, store, context) => {
  const html = (
    <Provider store={store}>
      <StaticRouter location={req.path} context={context}>
        <div className="app-container">{renderRoutes(AppRoutes)}</div>
      </StaticRouter>
    </Provider>
  );
  const content = renderToString(html);
  const helmet = Helmet.renderStatic();

  return `
    <!DOCTYPE html>
    <html>
      <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no" />

        ${helmet.title.toString()}
        ${helmet.meta.toString()}

        <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
        <link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
        <link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
        <link rel="manifest" href="/manifest.json">
        <link rel="mask-icon" href="/safari-pinned-tab.svg" color="#43578c">

        <!-- iOS -->
        <meta name="apple-mobile-web-app-capable" content="yes">
        <meta name="apple-mobile-web-app-status-bar-style" content="black">
        <meta name="apple-mobile-web-app-title" content="Geopulse">
        <link rel="apple-touch-startup-image" href="/assets/geopulse-splash.png">
      
        <!-- Other -->
        <meta name="application-name" content="Geopulse">
        <meta name="theme-color" content="#43578c">
        <meta name="format-detection" content="telephone=no">
      
        <!-- Social -->
        <meta itemprop="name" content="Geopulse">
        <meta itemprop="description" content="Get a pulse on the happenings nearby">
        <meta itemprop="image" content="https://geopulse-app.herokuapp.com/assets/geopulse-banner.png">
      
        <meta property="og:image" content="https://geopulse-app.herokuapp.com/assets/geopulse-banner.png">
        <meta property="og:image:width" content="1200">
        <meta property="og:image:height" content="628">
        <meta property="og:title" content="Geopulse">
        <meta property="og:description" content="Get a pulse on the happenings nearby">
        <meta property="og:url" content="https://geopulse-app.herokuapp.com">
      
        <meta name="twitter:card" content="summary_large_image">
        <meta name="twitter:title" content="Geopulse">
        <meta name="twitter:description" content="Get a pulse on the happenings nearby">
        <meta name="twitter:creator" content="@seriouslysean">
        <meta name="twitter:image" content="https://geopulse-app.herokuapp.com/assets/geopulse-banner.png">
        
        <link rel="stylesheet" type="text/css" href="bundle.css" />
      </head>
      <body>
        <div id="root">${content}</div>
        <script>window.INITIAL_STATE = ${serialize(store.getState())}</script>
        <script src="bundle.js"></script>
      </body>
    </html>
  `;

  res.send(html);
};
