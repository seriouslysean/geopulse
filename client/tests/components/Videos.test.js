import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Videos from "../../src/components/Videos";
import videosProp from "../../../data/fixtures/videos";

test("should show Videos loading correctly", () => {
  const wrapper = shallow(<Videos />);
  expect(wrapper).toMatchSnapshot();
});

test("should render Videos with content correctly", () => {
  const wrapper = shallow(<Videos ready={true} videos={videosProp} />);
  expect(wrapper).toMatchSnapshot();
});
