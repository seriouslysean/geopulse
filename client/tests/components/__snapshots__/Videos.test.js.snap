// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Videos with content correctly 1`] = `
<section
  className="section section--spacing section--bg-earth-darker videos"
>
  <div
    className="content-container content-container--grid-spacing"
  >
    <div
      className="videos__content"
    >
      <h1
        className="section__title"
      >
        Videos
      </h1>
      <div
        className="content"
      >
        <ul
          className="videos__list"
        >
          <VideosVideo
            key="2BsRw8wFjyI"
            video={
              Object {
                "etag": "\\"7991kDR-QPaa9r0pePmDjBEa2h8/jF8beQSw6HKOnICIbqGhzAcYQgc\\"",
                "id": Object {
                  "kind": "youtube#video",
                  "videoId": "2BsRw8wFjyI",
                },
                "kind": "youtube#searchResult",
                "snippet": Object {
                  "channelId": "UCBi2mrWuNuyYy4gbM6fU18Q",
                  "channelTitle": "ABC News",
                  "description": "Diane Sawyer spent months inside one of the country's most dangerous schools.",
                  "liveBroadcastContent": "none",
                  "publishedAt": "2013-12-05T10:16:36.000Z",
                  "thumbnails": Object {
                    "default": Object {
                      "height": 90,
                      "url": "https://i.ytimg.com/vi/2BsRw8wFjyI/default.jpg",
                      "width": 120,
                    },
                    "high": Object {
                      "height": 360,
                      "url": "https://i.ytimg.com/vi/2BsRw8wFjyI/hqdefault.jpg",
                      "width": 480,
                    },
                    "medium": Object {
                      "height": 180,
                      "url": "https://i.ytimg.com/vi/2BsRw8wFjyI/mqdefault.jpg",
                      "width": 320,
                    },
                  },
                  "title": "At Strawberry Mansion High, There's Fear, Hope",
                },
              }
            }
          />
          <VideosVideo
            key="kAmDOTsp5cs"
            video={
              Object {
                "etag": "\\"7991kDR-QPaa9r0pePmDjBEa2h8/sRarOmG5B--B9TcQ8QX5D7ESEUc\\"",
                "id": Object {
                  "kind": "youtube#video",
                  "videoId": "kAmDOTsp5cs",
                },
                "kind": "youtube#searchResult",
                "snippet": Object {
                  "channelId": "UClKgO_YP3pcIVeav2g_k6hw",
                  "channelTitle": "fightTIPS",
                  "description": "Get Shane's Jump Rope for less than $4 (FREE SHIPPING): http://full.sc/Y21zkh Jumping rope is a great warm-up exercise for any boxer, athlete, or just someone wanting to try a new workout!...",
                  "liveBroadcastContent": "none",
                  "publishedAt": "2014-06-08T21:30:01.000Z",
                  "thumbnails": Object {
                    "default": Object {
                      "height": 90,
                      "url": "https://i.ytimg.com/vi/kAmDOTsp5cs/default.jpg",
                      "width": 120,
                    },
                    "high": Object {
                      "height": 360,
                      "url": "https://i.ytimg.com/vi/kAmDOTsp5cs/hqdefault.jpg",
                      "width": 480,
                    },
                    "medium": Object {
                      "height": 180,
                      "url": "https://i.ytimg.com/vi/kAmDOTsp5cs/mqdefault.jpg",
                      "width": 320,
                    },
                  },
                  "title": "How to Jump Rope Like a Boxer",
                },
              }
            }
          />
          <VideosVideo
            key="ooMjhFToGQw"
            video={
              Object {
                "etag": "\\"7991kDR-QPaa9r0pePmDjBEa2h8/gazkb9kyX-JUuukjJVFfFJI46B4\\"",
                "id": Object {
                  "kind": "youtube#video",
                  "videoId": "ooMjhFToGQw",
                },
                "kind": "youtube#searchResult",
                "snippet": Object {
                  "channelId": "UClKgO_YP3pcIVeav2g_k6hw",
                  "channelTitle": "fightTIPS",
                  "description": "Subscribe to fightTIPS▻http://bit.ly/1APnzvw Online MMA Training Courses▻http://bit.ly/1FIOiju This tutorial video explains how to get faster head movement using various boxing drills,...",
                  "liveBroadcastContent": "none",
                  "publishedAt": "2016-10-05T00:43:52.000Z",
                  "thumbnails": Object {
                    "default": Object {
                      "height": 90,
                      "url": "https://i.ytimg.com/vi/ooMjhFToGQw/default.jpg",
                      "width": 120,
                    },
                    "high": Object {
                      "height": 360,
                      "url": "https://i.ytimg.com/vi/ooMjhFToGQw/hqdefault.jpg",
                      "width": 480,
                    },
                    "medium": Object {
                      "height": 180,
                      "url": "https://i.ytimg.com/vi/ooMjhFToGQw/mqdefault.jpg",
                      "width": 320,
                    },
                  },
                  "title": "Faster Head Movement: Boxing Drills for Dodging Punches",
                },
              }
            }
          />
          <VideosVideo
            key="26DSA5DRz0s"
            video={
              Object {
                "etag": "\\"7991kDR-QPaa9r0pePmDjBEa2h8/dJiKC1ukNumaD5aRD5JS11b5GKQ\\"",
                "id": Object {
                  "kind": "youtube#video",
                  "videoId": "26DSA5DRz0s",
                },
                "kind": "youtube#searchResult",
                "snippet": Object {
                  "channelId": "UClKgO_YP3pcIVeav2g_k6hw",
                  "channelTitle": "fightTIPS",
                  "description": "There's a good reason why the jab is the first punch you will learn in boxing: because it should be your BEST punch! It exerts the least amount of energy, acts as a probe, dictates the range,...",
                  "liveBroadcastContent": "none",
                  "publishedAt": "2017-01-01T19:41:46.000Z",
                  "thumbnails": Object {
                    "default": Object {
                      "height": 90,
                      "url": "https://i.ytimg.com/vi/26DSA5DRz0s/default.jpg",
                      "width": 120,
                    },
                    "high": Object {
                      "height": 360,
                      "url": "https://i.ytimg.com/vi/26DSA5DRz0s/hqdefault.jpg",
                      "width": 480,
                    },
                    "medium": Object {
                      "height": 180,
                      "url": "https://i.ytimg.com/vi/26DSA5DRz0s/mqdefault.jpg",
                      "width": 320,
                    },
                  },
                  "title": "5 Common Jab Mistakes: This Should be Your Best Punch!",
                },
              }
            }
          />
        </ul>
      </div>
      <a
        className="section__attribution"
        href="https://www.youtube.com/"
        rel="external"
        target="_blank"
      >
        Powered by YouTube
      </a>
    </div>
  </div>
</section>
`;

exports[`should show Videos loading correctly 1`] = `
<section
  className="section section--spacing section--bg-earth-darker videos"
>
  <div
    className="content-container content-container--grid-spacing"
  >
    <div
      className="videos__content"
    >
      <h1
        className="section__title"
      >
        Videos
      </h1>
      <Loading />
      <a
        className="section__attribution"
        href="https://www.youtube.com/"
        rel="external"
        target="_blank"
      >
        Powered by YouTube
      </a>
    </div>
  </div>
</section>
`;
